pkg/server/store/appdefault/store.go

```diff
@@ -17,7 +17,8 @@ func New(config *config.Config) *Store {
 	}
 }
 
-func (c *Store) List(_ *types.APIRequest, _ *types.APISchema) (types.APIObjectList, error) {
+func (c *Store) List(req *types.APIRequest, _ *types.APISchema) (types.APIObjectList, error) {
+	req.Type = "appdefaults"
```

Comment:
Could you elaborate on how setting the plural type on the **Request** object resolves the issue?

Reply:
Rancher Desktop has the `resourceType` hardcoded to `channels`: https://github.com/rancher-sandbox/rancher-desktop/pull/7461/commits/f1034a869db1e84d465484ba4f61095e627d17c1

Reply:
Hi @jandubois, thank you for the link. However, it doesn’t clarify why we need to set the plural type on the **Request** object. I thought it should be set on the response object, which doesn’t seem to exist in this context. Could it be that the `*types.APIRequest` is internally translated into the HTTP response object as part of how the rancher/api-server operates?

Reply:
Oh, I'm sorry, I misunderstood your question.

That would probably be best for @brandond to answer.

Reply:
OK, I think I figured it out: the value of `req.Type` is used to set the `ResourceType` field on the `GenericCollection` object. ([link](https://github.com/rancher/apiserver/blob/main/pkg/writer/encoding.go#L128))

Resolving this conversation.

Reply:
As noted in the linked k3s issue:
> Rancher Desktop 1.16 (the latest public release) checks the resourceType: https://github.com/rancher-sandbox/rancher-desktop/pull/7461

I'd have to ask @mook-as why it was decided that it was a good idea to check for this specific field value, as opposed to just checking that the `.data[]` had non-zero length.

Reply:
This was around the time of https://github.com/rancherlabs/eio/issues/2483 and I probably went overboard in trying to make sure the response was valid. Since there wasn't an obvious schema I just want with whatever the channel server responded at the time.

Reply:
Yeah, unfortunately the k3s channel server had been accidentally rolled back to v0.4.0 or thereabouts when you checked.